### ISSUE TITLE
🐛 Failed to lint json from stdin.

### ISSUE THREAD

**Issue by PFiS1737** [biomejs/biome#4102]
### Environment information

```block
CLI:
  Version:                      1.9.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-kitty"
  JS_RUNTIME_VERSION:           "v22.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.3" (In fact, i use pnpm 9.11.0)

Biome Configuration:
  Status:                       unset

Workspace:
  Open Documents:               0
```

### What happened?

1. open your terminal.
2. ```shell
   mkdir demo && cd demo
   pnpm init
   pnpm i -D @biomejs/biome

   # no error
   npx biome lint package.json

   # get "The contents aren't fixed. Use the `--fix` flag to fix them."
   # note that we have already set "--fix"
   cat package.json | npx biome lint --stdin-file-path=package.json --fix
   ```

### Expected result

When using stdin mode, it shouldn't return an error. (Shouldn't it behave the same as normal usage?)

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by guacs** [comment]
@dyc3, this issue exists for formatting as well. The fix included in #4109 only handles this for the lint and check command, and not for the format. You can see this issue on 1.9.4 by running the following:

`echo | biome format --write --stdin-file-path=foo.js`

Should this issue be reopened, or would you like me to open another issue for this?
